share-on-mastodon.js

```diff
@@ -197,120 +208,143 @@ export class shareOnMastodon extends HTMLElement {
   constructor() {
     super();
 
-    this.attachShadow({ mode: 'open' });
+    this.attachShadow({ mode: "open" });
     this.shadowRoot.appendChild(template.content.cloneNode(true));
 
     // Get button text from attribute if available
-    if (this.getAttribute('button_text')) {
-      this.shadowRoot.querySelector('#btnText').innerHTML = this.getAttribute('button_text');
-    };
+    if (this.getAttribute("data-button-text")) {
+      this.shadowRoot.querySelector("#button-text").innerHTML =
+        this.getAttribute("data-button-text");
+    }
 
     // Get the modal heading from attribute if available
-    if (this.getAttribute('modal_heading')) {
-      this.shadowRoot.querySelector('#modalHeading').innerHTML = this.getAttribute('modal_heading');
-    };
+    if (this.getAttribute("data-modal-heading")) {
+      this.shadowRoot.querySelector("#modal-heading").innerHTML =
+        this.getAttribute("data-modal-heading");
+    }
+
+    // Get the modal text from attribute if available
+    if (this.getAttribute("data-modal-text")) {
+      this.shadowRoot.querySelector("#modal-text").innerHTML =
+        this.getAttribute("data-modal-text");
+    }
 
     // Get the modal text from attribute if available
-    if (this.getAttribute('modal_text')) {
-      this.shadowRoot.querySelector('#modalText').innerHTML = this.getAttribute('modal_text');
-    };
+    if (this.getAttribute("data-jargon-text")) {
+      this.shadowRoot.querySelector("#instance-jargon").innerHTML =
+        this.getAttribute("data-jargon-text");
+    }
 
     // Get the default url to use as placeholder text from attribute if available
-    if (this.getAttribute('default_url')) {
-      this.shadowRoot.querySelector('#instanceInput').placeholder = this.getAttribute('default_url');
-    };
+    if (this.getAttribute("data-default-url")) {
+      this.shadowRoot.querySelector("#instance-input").placeholder =
+        this.getAttribute("data-default-url");
+    }
 
     // Get the share button text from attribute if available
-    if (this.getAttribute('share_text')) {
-      this.shadowRoot.querySelector('#shareText').innerHTML = this.getAttribute('share_text');
-    };
+    if (this.getAttribute("data-share-text")) {
+      this.shadowRoot.querySelector("#share-text").innerHTML =
+        this.getAttribute("data-share-text");
+    }
   }
 
   openModal() {
-  // Prefill the form with the user's previously-specified Mastodon instance, if applicable
-  if (localStorage['mastodon_instance']) {
-    this.shadowRoot.querySelector('#instanceInput').value = localStorage['mastodon_instance'];
-  }
+    // Prefill the form with the user's previously-specified Mastodon instance, if applicable
+    const storedInstance = localStorage.getItem("mastodonInstance");
```

Comment:
```suggestion
    const storedInstance = localStorage?.getItem("mastodonInstance");
```
Client side storage can be disabled in some cases.